src/Illuminate/Contracts/Encryption/Encrypter.php

```diff
@@ -7,43 +7,29 @@ interface Encrypter
     /**
      * Encrypt the given value.
      *
-     * @param  mixed  $value
-     * @param  bool  $serialize
-     * @return string
-     *
      * @throws \Illuminate\Contracts\Encryption\EncryptException
      */
-    public function encrypt(#[\SensitiveParameter] $value, $serialize = true);
+    public function encrypt(#[\SensitiveParameter] $value, bool $serialize = true, bool $deterministic = false): string;
 
     /**
      * Decrypt the given value.
      *
-     * @param  string  $payload
-     * @param  bool  $unserialize
-     * @return mixed
-     *
      * @throws \Illuminate\Contracts\Encryption\DecryptException
      */
-    public function decrypt($payload, $unserialize = true);
+    public function decrypt(string $payload, bool $unserialize = true): mixed;
```

Comment:
These changes to the contracts are major breaking changes, and cannot be accepted on 12.x.